docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,461 @@
+
+[float]
+==== Removing master-eligible nodes
+
+If you wish to remove some of the master-eligible nodes in your cluster, you
+must first reduce the `cluster.master_nodes_failure_tolerance` setting to match
+the target cluster size before removing the extraneous nodes. This temporary
+situation is the only case in which `cluster.master_nodes_failure_tolerance`
+should be set lower than the recommended values above.
+
+You must also be careful not to remove too many master-eligible nodes all at
+the same time. For instance, if you currently have seven master-eligible nodes
+and you wish to reduce this to three, you cannot simply stop four of the nodes
+all at the same time: to do so would leave only three nodes remaining, which is
+less than half of the cluster, which means it cannot take any further actions.
+You should remove the nodes one-at-a-time and verify that each node has been
+removed from the configuration before moving onto the next one, using the
+`await_removal` API:
+
+[source,js]
+--------------------------------------------------
+# Explicit timeout of one minute
+GET /_nodes/node_name/await_removal?timeout=1m
+# Default timeout of 30 seconds
+GET /_nodes/node_name/await_removal
+--------------------------------------------------
+// CONSOLE
+
+The node (or nodes) for whose removal to wait are specified using
+<<cluster-nodes,node filters>> in place of `node_name` here.
+
+A special case of this is the case where there are only two master-eligible
+nodes and you wish to remove one of them. In this case neither node can be
+safely shut down since both nodes are required to reliably make progress, so
+you must first explicitly _retire_ one of the nodes. A retired node still works
+normally, but Elasticsearch will try and transfer it out of the current
+configuration so its vote is no longer required, and will never move a retired
+node back into the configuration. Once a node has been retired and is removed
+from the configuration, it is safe to shut it down. A node can be retired using
+the retirement API:
+
+[source,js]
+--------------------------------------------------
+# Retire node and wait for its removal up to the default timeout of 30 seconds
+POST /_nodes/node_name/retire
+# Retire node and wait for its removal up to one minute
+POST /_nodes/node_name/retire?timeout=1m
+--------------------------------------------------
+// CONSOLE
+
+The node to retire is specified using <<cluster-nodes,node filters>> in place
+of `node_name` here. If a call to the retirement API fails then the call can
+safely be retried. However if a retirement fails then it's possible the node
+cannot be removed from the configuration due to the
+`cluster.master_nodes_failure_tolerance` setting, so verify that this is set
+correctly first. A successful response guarantees that the node has been
+removed from the configuration and will not be reinstated.
+
+A node (or nodes) can be brought back out of retirement using the `unretire`
```

Comment:
means we probably also need an API to figure out which nodes are retired.

Reply:
This should be available from the cluster state - I added a note about this.